documentation/bolt_defaults_reference.md.erb

```diff
@@ -0,0 +1,92 @@
+# `bolt-defaults.yaml` options
+
+This page lists the configuration options that are available in the `bolt-defaults.yaml` configuration file,
+which is supported in both the [system-wide and user-level configuration directories](configuring_bolt.md).
+The avaiable options allow you to configure how Bolt runs and set the default configuration for each of the
+available transports.
+
+Options that are set in a `bolt-defaults.yaml` file in the system-wide configuration directory will apply
+to all Bolt projects run on the system. Likewise, options set in a `bolt-defaults.yaml` file in the user-level
+configuration directory apply to all Bolt projects run by the user.
+
+## Top-level options
+
+These are the configuration options that can be set at the top-level of the file.
+
+<% @top_level[:options].each do |option, desc| -%>
+### `<%= option %>`
+
+<%= desc %>
+
+<% if @top_level[:defaults][option] -%>
+- **Default:** <%= @top_level[:defaults][option] %>
+<% end %>
+<% end %>
+
+## Additional options
+
+Several configuration options accept a map of configuration options.
+
+### `inventory-config`
+
+The top-level `inventory-config` option lets you to set the default transport used by targets that do
+not specify a transport in their URI or in the `inventory.yaml` file.
+
+<% @inventory[:options].each do |option, desc| -%>
+#### `<%= option %>`
+
+<%= desc %>
+
+<% if @inventory[:defaults][option] -%>
+- **Default:** <%= @inventory[:defaults][option] %>
+<% end %>
+<% end %>
+
+<% @transports[:options].each do |transport, options| -%>
```

Comment:
This lists all of the options for each transport. I'm of two minds on having those options in this reference doc vs having a separate 'transports reference' page:

- Explicitly listing the transport options for each file that uses them means users don't have to jump back and forth between pages to get the info they need.
- This adds a lot of redundancy between reference pages and potentially adds unnecessary clutter.

This might be something better suited for #1888 , but I wanted to see what it would look like here for feedback/discussion.